daemon/algod/api/server/v2/handlers.go

```diff
@@ -0,0 +1,420 @@
+	ledger := v2.Node.Ledger()
+	latestBlkHdr, err := ledger.BlockHdr(ledger.Latest())
+	if err != nil {
+		return returnError(ctx, http.StatusInternalServerError, err, errFailedRetrievingNodeStatus, v2.Log)
+	}
+
+	// TODO: Replace this with Node.Status().StoppedAtUnsupportedRound ?
+	if latestBlkHdr.NextProtocol != "" {
+		if _, nextProtocolSupported := config.Consensus[latestBlkHdr.NextProtocol]; !nextProtocolSupported {
+			// see if the desired protocol switch is expect to happen before or after the above point.
+			if latestBlkHdr.NextProtocolSwitchOn <= basics.Round(round+1) {
+				// we would never reach to this round, since this round would happen after the (unsupported) protocol upgrade.
+				return returnError(ctx, http.StatusBadRequest, err, errRequestedRoundInUnsupportedRound, v2.Log)
+			}
+		}
+	}
+
+	// Wait
+	select {
+	case <-v2.Shutdown:
+		return returnError(ctx, http.StatusInternalServerError, err, errServiceShuttingDown, v2.Log)
+	case <-time.After(1 * time.Minute):
+	case <-ledger.Wait(basics.Round(round + 1)):
+	}
+
+	// Return status after the wait
+	return v2.GetStatus(ctx)
+}
+
+// RawTransaction broadcasts a raw transaction to the network.
+// (POST /v2/transactions)
+func (v2 *Handlers) RawTransaction(ctx echo.Context) error {
+	var txgroup []transactions.SignedTxn
+	dec := protocol.NewDecoder(ctx.Request().Body)
+	for {
+		var st transactions.SignedTxn
+		err := dec.Decode(&st)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return returnError(ctx, http.StatusBadRequest, err, err.Error(), v2.Log)
+		}
+		txgroup = append(txgroup, st)
+	}
+
+	if len(txgroup) == 0 {
+		err := errors.New("empty txgroup")
+		return returnError(ctx, http.StatusBadRequest, err, err.Error(), v2.Log)
+	}
+
+	err := v2.Node.BroadcastSignedTxGroup(txgroup)
+	if err != nil {
+		return returnError(ctx, http.StatusBadRequest, err, err.Error(), v2.Log)
+	}
+
+	// For backwards compatibility, return txid of first tx in group
+	txid := txgroup[0].ID()
+	return ctx.JSON(http.StatusOK, generated.PostTransactionsResponse{TxId: txid.String()})
+}
+
+// TransactionParams gets parameters for constructing a new transaction
+// (GET /v2/transactions/params)
+func (v2 *Handlers) TransactionParams(ctx echo.Context) error {
+	stat, err := v2.Node.Status()
+	if err != nil {
+		return returnError(ctx, http.StatusInternalServerError, err, errFailedRetrievingNodeStatus, v2.Log)
+	}
+
+	gh := v2.Node.GenesisHash()
+
+	var params generated.TransactionParams
+	params.Fee = v2.Node.SuggestedFee().Raw
+	params.GenesisId = v2.Node.GenesisID()
+	params.GenesisHash = gh[:]
+	params.LastRound = uint64(stat.LastRound)
+	params.ConsensusVersion = string(stat.LastVersion)
+
+	proto := config.Consensus[stat.LastVersion]
+	params.MinFee = proto.MinTxnFee
+
+	return ctx.JSON(http.StatusOK, params)
+}
+
+// PendingTransactionInformation gets a specific pending transaction, or looks it up in the ledger if the recently
```

Comment:
nit - fix the description the match the function. Including the number of previous rounds it searches at (MaxTxnLife == 1000), etc.

Reply:
something is still off about this comment. The handler doesn't get any specific pending transaction. the handler takes a TXID and returns the associated transaction's information if found in the Transaction Pool or the last 1000 valid round.

Reply:
The comment reflects what happens inside `v2.Node.GetPendingTransaction(txID)`, which looks for the specified txID in the transaction pool, and if not found it looks for it in the last `proto.MaxTxnLife` rounds.

Reply:
> looks for the specified txID in the transaction pool, and if not found it looks for it in the last proto.MaxTxnLife rounds

This is a much better comment for this handler.